src/main/java/com/forgeessentials/signtools/SignToolsModule.java

```diff
@@ -120,35 +117,45 @@ public void onPlayerInteract(PlayerInteractEvent event)
 
             }
 
-            String[] signText = getFormatted(sign.signText);
+            String[] signText = getUnFormatted(sign.signText);
 
             if (APIRegistry.scripts.runEventScripts(signinteractKey, event.getEntityPlayer(), new SignInfo(event.getEntityPlayer().dimension, event.getPos(), signText, event)))
             {
                 event.setCanceled(true);
             }
 
+            /**
+             * how to use: First line of the sign MUST BE [command] Second line is the command you want to run Third and fourth
+             * lines are arguments to the command.
+             */
             if (allowSignCommands && (event instanceof RightClickBlock))
             {
-                if (signText[0].equals("[command]"))
+                // if (signText[0].trim() == "[command]")
+                if (signText[0].trim().equalsIgnoreCase("[command]"))
                 {
-                    String send = signText[1] + " " + signText[2] + " " + signText[3];
-                    if (send != null && FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager() != null)
+                    if (PermissionAPI.hasPermission(event.getEntityPlayer(), USESIGNCOMMANDS_PERM))
                     {
-                        FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager().executeCommand(event.getEntityPlayer(), send);
-                        event.setCanceled(true);
+                        String send = signText[1] + " " + signText[2] + " " + signText[3];
+                        if (send != null && FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager() != null)
+                        {
+                            FMLCommonHandler.instance().getMinecraftServerInstance().getCommandManager().executeCommand(event.getEntityPlayer(), send);
+                            event.setCanceled(true);
+                        }
                     }
+                    else
+                        ChatOutputHandler.chatError(event.getEntityPlayer(), Translator.translate("You are not allowed to use sign commands!"));
                 }
             }
 
         }
 
     }
 
-    private String[] getFormatted(ITextComponent[] text)
+    private String[] getUnFormatted(ITextComponent[] text)
     {
         String[] out = new String[text.length];
         for (int i = 0; i < text.length; i++) {
-            out[i] = text[i].getFormattedText().replace(ChatOutputHandler.COLOR_FORMAT_CHARACTER, '&');
+            out[i] = text[i].getUnformattedText().replace(ChatOutputHandler.COLOR_FORMAT_CHARACTER, '&');
```

Comment:
Again, why is this changing to unformatted text.  What do you wish to solve by doing this?